localstack/services/lambda_/invocation/event_manager.py

```diff
@@ -7,34 +7,277 @@
+    ) -> ReceiveMessageResult:
+        queue = QUEUE_MANAGER.get_queue(queue_name=QueueUrl)
+        num = MaxNumberOfMessages or 1
+        result = queue.receive(
+            num_messages=num,
+            visibility_timeout=VisibilityTimeout,
+            wait_time_seconds=WaitTimeSeconds,
+        )
+
+        messages = []
+        for i, standard_message in enumerate(result.successful):
+            message = to_sqs_api_message(standard_message, attribute_names, MessageAttributeNames)
+            message["ReceiptHandle"] = result.receipt_handles[i]
+            messages.append(message)
+
+        return {"Messages": messages if messages else None}
+
+    def delete_message(self, QueueUrl: String, ReceiptHandle: String) -> None:
+        queue = QUEUE_MANAGER.get_queue(queue_name=QueueUrl)
+        queue.remove(ReceiptHandle)
+
+    def _create_message_attributes(
+        self,
+        message_system_attributes: MessageBodySystemAttributeMap = None,
+    ) -> dict[str, str]:
+        result = {
+            MessageSystemAttributeName.SenderId: config.INTERNAL_RESOURCE_ACCOUNT,  # not the account ID in AWS
+            MessageSystemAttributeName.SentTimestamp: str(now(millis=True)),
+        }
+
+        if message_system_attributes is not None:
+            for attr in message_system_attributes:
+                result[attr] = message_system_attributes[attr]["StringValue"]
+
+        return result
+
+    def send_message(
+        self,
+        QueueUrl: String,
+        MessageBody: String,
+        DelaySeconds: Integer = None,
+        MessageAttributes: MessageBodyAttributeMap = None,
+        MessageSystemAttributes: MessageBodySystemAttributeMap = None,
+        MessageDeduplicationId: String = None,
+        MessageGroupId: String = None,
+    ) -> SendMessageResult:
+        queue = QUEUE_MANAGER.get_queue(queue_name=QueueUrl)
+
+        message = Message(
+            MessageId=generate_message_id(),
+            MD5OfBody=md5(MessageBody),
+            Body=MessageBody,
+            Attributes=self._create_message_attributes(MessageSystemAttributes),
+            MD5OfMessageAttributes=_create_message_attribute_hash(MessageAttributes),
+            MessageAttributes=MessageAttributes,
+        )
+        queue_item = queue.put(
+            message=message,
+            message_deduplication_id=MessageDeduplicationId,
+            message_group_id=MessageGroupId,
+            delay_seconds=int(DelaySeconds) if DelaySeconds is not None else None,
+        )
+        message = queue_item.message
+        return {
+            "MessageId": message["MessageId"],
+            "MD5OfMessageBody": message["MD5OfBody"],
+            "MD5OfMessageAttributes": message.get("MD5OfMessageAttributes"),
+            "SequenceNumber": queue_item.sequence_number,
+            "MD5OfMessageSystemAttributes": _create_message_attribute_hash(MessageSystemAttributes),
+        }
+
+
+CLIENT = FakeSqsClient()
```

Comment:
maybe we can use the factory pattern here instead to defer the creation
```suggestion
@singleton_factory
def get_fake_sqs_client():
    return FakeSqsClient(QueueManager())
```

Reply:
Yeah, that was a quick hack, good catch, would have missed it!